containers/Investment/RevolvingPool/View/InvestCard.tsx

```diff
@@ -2,19 +2,49 @@ import * as React from 'react'
 import { Box, Button } from 'grommet'
 import { TokenInput } from '@centrifuge/axis-token-input'
 import { Pool } from '../../../../config'
+import { createTransaction, useTransactionState, TransactionProps } from '../../../../ducks/transactions'
+import { ITinlake as ITinlakeV3 } from '@centrifuge/tinlake-js-v3'
+import { connect } from 'react-redux'
 
 import { Description } from './styles'
 import { Card } from './TrancheOverview'
 
-interface Props {
+interface Props extends TransactionProps {
   pool: Pool
   tranche: 'senior' | 'junior'
   setCard: (card: Card) => void
+  tinlake: ITinlakeV3
 }
 
 const InvestCard: React.FC<Props> = (props: Props) => {
   const token = props.tranche === 'senior' ? 'DROP' : 'TIN'
-  const [daiValue, setDaiValue] = React.useState('234000000000000000')
+  const [daiValue, setDaiValue] = React.useState('0')
+
+  const [limit, setLimit] = React.useState<string | undefined>(undefined)
+
+  React.useEffect(() => {
+    async function getLimit() {
+      const user = await props.tinlake.signer?.getAddress()
+      if (user) {
+        const balance = await props.tinlake.getCurrencyBalance(user)
+        setLimit(balance.toString())
+      }
+    }
+    getLimit()
+  }, [props.tinlake])
```

Comment:
Can we not use an async function right away?

```suggestion
  React.useEffect(async () => {
    const user = await props.tinlake.signer?.getAddress()
    if (user) {
      const balance = await props.tinlake.getCurrencyBalance(user)
      setLimit(balance.toString())
    }
  }, [props.tinlake])
```

Reply:
`async` handlers are not allowed for `useEffect` 😞 https://github.com/facebook/react/issues/14326#issuecomment-441680293